File: implement-shell-tools/cat/cat*.js
const fs = require("fs");

const data = process.argv.slice(4);

for (const file of data) {
  try {
    const content = fs.readFileSync(file, "utf8");
    process.stdout.write(content);
  } catch (err) {
    console.error(`cat: ${file}: No file available`);
  }
}

File: implement-shell-tools/cat/cat*.py
from pathlib import Path
import sys

try: 
    data = sys.argv[3:]

    for item in data:
        content = Path(item).read_text(encoding="utf-8")
        print(content, end="")
except FileNotFoundError:
        sys.stderr.write(f"cat: {item}: No such file or directory\n")